public/app/features/dashboard-scene/panel-edit/VizPanelManager.tsx

```diff
@@ -346,10 +348,31 @@ export class VizPanelManager extends SceneObjectBase<VizPanelManagerState> {
     dataObj.runQueries();
   }
 
-  public changeQueries(queries: DataQuery[]) {
-    const dataObj = this.queryRunner;
-    dataObj.setState({ queries });
-    // TODO: Handle dashboard query
+  public changeQueries<T extends DataQuery>(queries: T[]) {
+    const dataObj = this.state.panel.state.$data;
+    const runner = this.queryRunner;
+
+    // Updating Dashboard query requires a special treatment as the SharedDataProvider in edit mode contains
```

Comment:
Why does the SharedDataProvider have a clone of the source panel data provider?

Reply:
@torkelo the reason is not obvious at first sight, but you can reason about it when you look at the `ShareQueryDataProvider` implemention and how it resolves the source panel: https://github.com/grafana/grafana/blob/dash-scene-dash-datasource/public/app/features/dashboard-scene/scene/ShareQueryDataProvider.ts#L71

It depends on resolving the scene root, which for the new panel edit is PanelEditPage, which does not contain other panels. It works for the DashbaordScenePage that contains all the viz panels, but not for the panel edit.

I think this shows other problem which is the routes inconsistencies between old and new architecture. The new arch panel edit route contains `/panel-edit/XXX` segment in the url while the old arch depend on the query param only.

Reply:
Well, technically we could also use the dashboard ref that's passed to the `VizPanelManager` but then we would have to manually activate corresponding source panel query runners, which sounds less optimal than just relying on the library activation flow that's executed when we attach a clone to the SharedQueryDataProvider...

Reply:
ah, I understand. dam it.

I could at first not think of any good alternative but I think I have an idea. 

What if ShareQueryDataProviderState had a lookupSource? callback that we can set. Nah second thought that is feels like it could be equally complex. 

Another idea is for ShareQueryDataProvider _subscribeToSource, extract a new function named _findSourcePanel where it would check if the root is PanelEditor and if that is the case get the dashboard from it and do the lookup as usual.

Reply:
I think that last solution (of making ShareQueryDataProvider aware that root can be PanelEditor) looks pretty simple and should avoid some of this complexity of cloning the source

Reply:
@torkelo it sounds easy but it's a bit more complex in reality. The source panel exists in a scene (dashboard scene) that's not active. It has a different time range context. So while subscribing to the viz panel from another scene would work if the root is PanelEditor, the changes to the i.e. time range will not affect the panel in the edit mode. The missing piece is parent wiring. 

Did a quick sketch of the solution and this does not work unfortunately. I mean - it does work and renders the panel, but the problem is that we:
- still need to setup time range subscription in order to react to a different time range scope in the PanelEditor
- would have to change SceneQueryRunner API in order to make `runWithTimeRange` public so that there's possiblity to provide time range to manually executed query.

Reply:
Ah, did not think of time range (and variables) they would all subscribe to the dashboard not panel edit state

Reply:
is there a way to move the copy / clone of the source data provider into ShareQueryDataProvider (when it detects PanelEditor root) ?

Reply:
> is there a way to move the copy / clone of the source data provider into ShareQueryDataProvider (when it detects PanelEditor root) ?

Yes, that should be doable, think this is an interesting apprach, will try this

Reply:
Think it worked, pushed an update in [d9cf4cb](https://github.com/grafana/grafana/pull/79991/commits/d9cf4cbc078a535f5f8f26ea4168d7511e3d4d7a)